component-studio/component-runtime-di/src/test/java/org/talend/sdk/component/runtime/di/schema/TaCoKitGuessSchemaTest.java

```diff
@@ -119,7 +137,50 @@ public boolean guessSchemaThroughAction() {
             };
             final IllegalStateException exception =
                     Assertions.assertThrows(IllegalStateException.class, guessSchema::guessInputComponentSchema);
-            Assertions.assertEquals(EXPECTED_ERROR_MESSAGE, exception.getMessage());
+            assertEquals(EXPECTED_ERROR_MESSAGE, exception.getMessage());
+        }
+    }
+
+    @Test
+    void guessProcessorSchema() throws Exception {
+        try (final ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();
+                PrintStream out = new PrintStream(byteArrayOutputStream)) {
+            final Entry f1 = factory.newEntryBuilder()
+                    .withName("f1")
+                    .withType(Schema.Type.STRING)
+                    .build();
+            final Entry f2 = factory.newEntryBuilder()
+                    .withName("f2")
+                    .withType(Schema.Type.LONG)
+                    .withDefaultValue(11l)
+                    .build();
+            final Entry f3 = factory.newEntryBuilder()
+                    .withName("f3")
+                    .withType(Schema.Type.BOOLEAN)
+                    .build();
+            final Schema schema = factory.newSchemaBuilder(Schema.Type.RECORD)
+                    .withProp("aprop", "a property!")
+                    .withEntry(f1)
+                    .withEntry(f2)
+                    .withEntry(f3)
+                    .build();
+            Map<String, String> config = new HashMap<>();
+            config.put("configuration.param1", "parameter one");
+            config.put("configuration.param2", "parameter two");
+            final TaCoKitGuessSchema guessSchema = new TaCoKitGuessSchema(
+                    out, config, "test-classes", "TaCoKitGuessSchema",
+                    "outputDi", null, "1");
+            guessSchema.guessProcessorComponentSchema(schema, "out");
+            guessSchema.close();
+            final Pattern pattern = Pattern.compile("^\\[\\s*(INFO|WARN|ERROR|DEBUG|TRACE)\\s*]");
+            final String lines = Arrays.stream(byteArrayOutputStream.toString().split("\n"))
+                    .filter(l -> !pattern.matcher(l).find()) // filter out logs
+                    .filter(l -> l.startsWith("[") || l.startsWith("{")) // ignore line with non json data
+                    .collect(joining("\n"));
+            final String expected =
+                    "[{\"label\":\"f1\",\"length\":0,\"nullable\":false,\"originalDbColumnName\":\"f1\",\"precision\":0,\"talendType\":\"id_String\"},{\"default\":\"11\",\"defaut\":\"11\",\"label\":\"f2\",\"length\":0,\"nullable\":false,\"originalDbColumnName\":\"f2\",\"precision\":0,\"talendType\":\"id_Long\"},{\"label\":\"f3\",\"length\":0,\"nullable\":false,\"originalDbColumnName\":\"f3\",\"precision\":0,\"talendType\":\"id_Boolean\"},{\"comment\":\"branch name\",\"label\":\"out\",\"length\":0,\"nullable\":false,\"originalDbColumnName\":\"out\",\"precision\":0,\"talendType\":\"id_String\"}]";
```

Comment:
please rebase the code lastest, the assert may fail

Reply:
funny, made same changes than you! 👍🏻